src/machine/machine_rp2040_flash.go

```diff
@@ -13,6 +13,33 @@ func EnterBootloader() {
 	enterBootloader()
 }
 
+// HardwareID returns an identifier that is unique within
+// a particular chipset.
+//
+// The identity is one burnt into the MCU itself, or the
+// flash chip at time of manufacture.
+//
+// It's possible that two different vendors may allocate
+// the same HardwareID, so callers should take this into
+// account if needing to generate a globally unique id.
+//
+// The length of the hardware ID is vendor-specific, but
+// 8 bytes (64 bits) is common.
+func HardwareID() []byte {
+	// 13 = 1 + FLASH_RUID_DUMMY_BYTES(4) + FLASH_RUID_DATA_BYTES(8)
+	txBuf := make([]byte, 13)
```

Comment:
Should also do something about these heap allocations.

Reply:
i was less sure about these - i'll think about it.  It's a bit messier given the flash 'protocol stuff' and returning an unaligned slice into it.  I'm wondering if escape analysis would take care of allocating rxBuf on the stack?

Reply:
Looks like we can use just one buffer for tx and rx and return a slice of that, so I've done that.